### DISCUSSION TITLE
Tauri and ruffle for a desktop flash app

### DISCUSSION THREAD

**Discussion by Alfredoes234** [ruffle-rs/ruffle#15211] [Ideas]
I was thinking if it could be possible to have some sort of [Tauri](https://tauri.app/) plugin or something similar to allow for a flash (ruffle) app on desktop that supports multi-platforms, and could be very useful to create performant apps that aren't built on electron also to port flash games to desktop.

**Comment by Dinnerbone** [comment]
Ruffle already has a native desktop build, I'm not sure I can see the advantage to wrapping the web version as it would come with a large number of downsides compares to just using the native desktop version directly - for example, web cannot use sockets or filesystem or gpu outside of webgl2 specifications.

Is there a specific use case you have in mind that I'm missing here?

**Reply by Alfredoes234** [reply]
It's just I was looking at club penguin re-written and they use electron for their desktop app with pepper flash player. So I wondered what the plausibility would be for extenal apps or developers embeding it in two arguably better rust based systems.

**Reply by Alfredoes234** [reply]
With tauri being more memory efficient, due to not spawning multiple instances, lower package size as well and ruffle being a good emulator due to rusts memory management and no garbage collector.

**Reply by Alfredoes234** [reply]
Although this might not be a good situation to use ruffle, I haven't used it for long so I haven't grasped the full features and core of ruffle so tell me If I am wrong to think to use it here.

**Reply by adrian17** [reply]
> With tauri being more memory efficient, due to not spawning multiple instances, lower package size as well and ruffle being a good emulator due to rusts memory management and no garbage collector.

Assuming the Tauri app would wrap the web release, it'll inherently be less efficient than native, both in terms of CPU and GPU work.
(and not sure what you mean by "not spawning multiple instances")

**Reply by Alfredoes234** [reply]
[It's described here](https://stackoverflow.com/a/59225275) where it spawns sub processes whereas I'm pretty sure tauri does not spawn multiple sub instances.

**Reply by adrian17** [reply]
Okay, but you're comparing it with Electron, which we aren't using at all. I don't disagree that Tauri might be more efficient than Electron, but that's still going to be worse than the current native player we have :(

**Reply by Alfredoes234** [reply]
That wasn't my point I'm not trying to get rid of the current player just thinking about if you could have a tauri plugin for flash games that have a desktop platform because most use electron (With pepper flash player from what I have seen) I just thought it would be cool if there was a possibility for the same sorta thing for ruffle. But idk just thinking that's why this is in the ideas part of discussion. With examples of the individual desktop programs such as [bin weevils-rewritten](https://play.binweevils.net/), [new club penguin](https://newcp.net/download) and [moshi monsters rewritten](https://moshirewritten.com/download/). These all have their own desktop app because it allows for greater control. I even asked the bin weevils developers about support for ruffle and they said maybe in the future when the support is fully done.
![image](https://github.com/ruffle-rs/ruffle/assets/114923333/2df70923-1545-4531-935f-6d94c0ef56a2)
